### PR TITLE
Awesome books: plain JavaScript with objects

### PR DESCRIPTION
**Pull request by zmekonnen251**
## Description

In this PR, we built a mobile menu for [template 2](https://www.figma.com/file/l7SqJ3ZfkAKih9sFxvWSR4/Microverse-Student-Project-1?node-id=39%3A122) using **JavaScript** , **CSS** and **HTML**.

### General requirements

- [x] There are no linter errors.
- [x] Followed the **Github flow**.
- [x] The documentation is professional.

### JavaScript requirements

- [x] Followed list of [best practices for JavaScript](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md).

### Project requirements

- Implemented a basic UI with plain HTML:
- Created a collection that keeps a list of books (hint: you can use an array of objects for that).
- Created a function to add a new book to the collection, with title and author.
- Created a function to remove a book from the collection (hint: you can use the [array filter()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter) method).
- Display all books saved in the collection in the top part of the page.
- when a user clicks the "Add" button:
  - A new book is added to the collection.
  - The new book is displayed in the page.
- when a user clicks the "Remove" button:
  - The correct book is removed from the collection.
  - The correct book disappears from the page.
- Data is preserved in the browser's memory by using [localStorage](https://github.com/microverseinc/curriculum-html-css/blob/main/local_storage.md).
- Used only plain JavaScript.

Thank you for taking the time to review this pull request.

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  event.preventDefault();
+  const newBook = {
+    title: titleField.value,
+    author: authorField.value,
+  };
+  awsomeBooks.push(newBook);
+  localStorage.setItem('storedBook', JSON.stringify(awsomeBooks));
+
+  addBookToPage(newBook);
+});
```
Comment (medaminedev66) [inline script.js]: - [x] The data is not preserved in the `localstorage` after clicking on `Add` button. Please make sure that data is preserved in the local storage by using the dev tools:
image

File: script.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
+// console.log(document.forms[0]["add"]);
```
Comment (medaminedev66) [inline script.js]: - [x] Please remove the commented code to make your code clean.

File: script.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+  newBook.appendChild(pTitle);
+  newBook.appendChild(pAuthor);
+  newBook.appendChild(button);
+  newBook.appendChild(hr);
+
+  bookContainer.appendChild(newBook);
+
+  //   console.log(awsomeBooks);
+  //   awsomeBooks.push({ title: titleField.value, author: authorField.value });
```
Comment (medaminedev66) [inline script.js]: - [x] Please remove the commented code to make your code clean.

File: script.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+const removeBook = (event) => {
+  if (event.target.classList.contains('remove')) {
+    const bookToBeRemoved = awsomeBooks.find(
+      (book) => book.title === event.target.parentElement.firstChild.textContent,
+    );
+
+    awsomeBooks.splice(awsomeBooks.indexOf(bookToBeRemoved), 1);
+    localStorage.setItem('storedBook', JSON.stringify(awsomeBooks));
+
+    // const storedBooks = JSON.parse(localStorage.getItem("storedBook"));
```
Comment (medaminedev66) [inline script.js]: - [x] Please remove the commented code to make your code clean.

File: script.js [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    bookContainer.innerHTML = '';
+    awsomeBooks.forEach((book) => {
+      addBookToPage(book);
+    });
+
+    // location.reload();
```
Comment (medaminedev66) [inline script.js]: - [x] Please remove the commented code to make your code clean.

File: README.md [hunk 6]
```diff
@@ -1 +1,36 @@
...
+## Built With
+
+- HTML and CSS
+- GitHub and Figma
+
+## Author
+
+- GitHub: [GitHub](https://github.com/zmekonnen251)
+- Twitter: [Twitter](https://twitter.com/mek_zela)
+- LinkedIn: [Zelalem G. Mekonnen](https://www.linkedin.com/in/zelalem-getachew/)
```
Comment (medaminedev66) [inline README.md:18]: - [ ] I see that two developers are working on this project. Kindly add the second developer info to the readme file.

File: README.md [hunk 7]
```diff
@@ -1 +1,36 @@
...
+
+Feel free to check the [issues page](../../issues/).
+
+## Show your support
+
+Give a ⭐️ if you like this project!
+
+## 📝 License
+
+This project is [MIT](./MIT.md) licensed.
```
Comment (medaminedev66) [inline README.md:44]: - [x] The licensed link is broken. Kindly fix it to make your README professional.

**Review by medaminedev66 — CHANGES_REQUESTED** [review]
Hi @student, :wave:

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight
- ✔️Git/GitHub flow best practices have been followed
- ✔️You have a descriptive PR

## Required Changes ♻️

_See the inline comments below for more required changes_

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ZeenatLawal — APPROVED** [review]
Hi @zmekonnen251 and @ahangarha, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

## Congratulations! 🎉 
6vEo


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**